### ISSUE TITLE
Rethink building SNI for multizone MeshService

### ISSUE THREAD

**Issue by jakubdyszkiewicz** [kumahq/kuma#10469]
### Description

The current way is a bit hacky
https://github.com/kumahq/kuma/pull/10451#discussion_r1637766609
While it works, we should consider something more robust.

Options that come to my mind:
1) More labels
2) Put SNI in MeshService and sync it cross zone. This is nice but how to handle splits?

xref https://github.com/kumahq/kuma/issues/10435

**Comment by jakubdyszkiewicz** [comment]
xref https://github.com/kumahq/kuma/pull/10918

**Comment by lahabana** [comment]
@jakubdyszkiewicz does this cover the implem of MADR-59?

**Comment by jakubdyszkiewicz** [comment]
yes, it's basically implem of https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/059-sni-in-the-resource.md

**Comment by lahabana** [comment]
> Put SNI in MeshService and sync it cross zone. This is nice but how to handle splits?
Do we agree that we decided to not do splits

**Comment by jakubdyszkiewicz** [comment]
> Do we agree that we decided to not do splits?

Yes, we do not do splits anymore

**Comment by lahabana** [comment]
The code for computing the SNI is in: https://github.com/kumahq/kuma/blob/4d8c12d42d9e5ca435eb6c4682786e57e016559c/pkg/core/resources/apis/meshservice/api/v1alpha1/helpers.go#L74-L95
